domain/personallearningplan/src/test/kotlin/uk/gov/justice/digital/hmpps/domain/personallearningplan/service/GoalServiceTest.kt

```diff
@@ -247,11 +247,13 @@ class GoalServiceTest {
       val prisonNumber = aValidPrisonNumber()
       val goalReference = aValidReference()
       val archiveGoal = aValidArchiveGoalDto(goalReference)
-      given(goalPersistenceAdapter.getGoal(prisonNumber, goalReference)).willReturn(null)
+      given(goalPersistenceAdapter.getGoal(any(), any())).willReturn(null)
```

Comment:
Our general pattern/approach with mockito is to setup our `given`s with `any()` args; and then `verify` the call was made with the expected args.
This is something we (me and the previous dev team) have done for years. The thinking being that if you setup `given`s with specific args and then the implementation changes in some way in respect of the args that the method is called with, you'll (hopefully) end up with a broken test (which is good), but it'll be broken most likely because of an NPE or similar which can be tricky to track down.
If you setup the `given`s with `any` arg matchers, then the call to the dependency will return the expected data no matter if the implementation changes in some way in respect of the args that the method is called with
So the test will pass down to the main `assertThat` (ie. asserting the return value from the method under test)
Then when we call `verify` on the mocks with the specific expected args, we'll get a clear failure message (and the test will fail) if the mock was called with different args